OpenRA.Mods.Common/Effects/Missile.cs

```diff
@@ -22,40 +22,98 @@ namespace OpenRA.Mods.Common.Effects
 {
 	class MissileInfo : IProjectileInfo
 	{
+		[Desc("Name of the image containing the projectile sequence.")]
 		public readonly string Image = null;
+
+		[Desc("Projectile sequence name.")]
 		[SequenceReference("Image")] public readonly string Sequence = "idle";
+
+		[Desc("Palette used to render the projectile sequence.")]
 		public readonly string Palette = "effect";
+
+		[Desc("Should the projectile's shadow be rendered?")]
 		public readonly bool Shadow = false;
+
 		[Desc("Projectile speed in WDist / tick")]
-		public readonly WDist Speed = new WDist(8);
-		[Desc("Maximum vertical pitch when changing altitude.")]
-		public readonly WAngle MaximumPitch = WAngle.FromDegrees(30);
+		public readonly WDist Speed = new WDist(8); // Rename to InitialSpeed
+
+		[Desc("Maximum projectile speed in WDist / tick")]
+		public readonly WDist MaximumSpeed = new WDist(512);
+
+		[Desc("Projectile acceleration when propulsion activated.")]
+		public readonly WDist Acceleration = WDist.Zero;
+
 		[Desc("How many ticks before this missile is armed and can explode.")]
 		public readonly int Arm = 0;
+
 		[Desc("Is the missile blocked by actors with BlocksProjectiles: trait.")]
 		public readonly bool Blockable = true;
+
 		[Desc("Maximum offset at the maximum range")]
 		public readonly WDist Inaccuracy = WDist.Zero;
+
 		[Desc("Probability of locking onto and following target.")]
 		public readonly int LockOnProbability = 100;
-		[Desc("In n/256 per tick.")]
-		public readonly int RateOfTurn = 5;
-		[Desc("Explode when following the target longer than this many ticks.")]
+
+		[Desc("Horizontal rate of turn.")]
+		public readonly int RateOfTurn = 5; // Rename to HorizontalRateOfTurn
+
+		[Desc("Vertical rate of turn.")]
+		public readonly int VerticalRateOfTurn = 5;
+
+		[Desc("Vertical launch angle (pitch).")]
+		public readonly WAngle LaunchAngle = WAngle.Zero;
+
+		[Desc("Run out of fuel after being activated this many ticks. Zero for unlimited fuel.")]
 		public readonly int RangeLimit = 0;
-		[Desc("Trail animation.")]
-		public readonly string Trail = null;
-		[Desc("Interval in ticks between each spawned Trail animation.")]
-		public readonly int TrailInterval = 2;
-		public readonly string TrailPalette = "effect";
-		public readonly bool TrailUsePlayerPalette = false;
+
+		[Desc("Explode when running out of fuel.")]
+		public readonly bool ExplodeWhenEmpty = true;
+
+		[Desc("Altitude above terrain below which to explode. Zero effectively deactivates airburst.")]
+		public readonly WDist AirburstAltitude = WDist.Zero;
+
+		[Desc("Cruise altitude. Zero means no cruise altitude used.")]
+		public readonly WDist CruiseAltitude = new WDist(512);
+
+		[Desc("Activate homing mechanism after this many ticks.")]
+		public readonly int ActivationDelay = 0;
+
+		[Desc("Image that contains the smoke animation.")]
+		public readonly string Trail = null; // Rename to SmokeImage
```

Comment:
The current code have them mixed - new tags and the code use Smoke* already with only the old tags aren't renamed - this should be reverted back entirely to use Trail*.